docs/it/refactoring.md

```diff
@@ -7,3 +7,132 @@
 # Altre idee
 
 - Differenza (Di intenti oltre che di tempo ed effort) tra refactoring automatico e manuale
+
+## Che cos'è il refactoring
+
+Alcuni direbbero che il refactoring è l'arte di crearsi problemi.
+Il refactoring è un insieme di attività dedite a rivedere una o più parti del codice con cui è composta la nostra applicazione.
+Può interessare un singolo metodo, una singola classe, un intero modulo.
+L'obiettivo di queste attività è rivedere la logica piuttosto che la scrittura del codice che la compone, in modo da ottenere migliori performance (o un maggior disaccoppiamento, una maggiore testabilità) senza però, in alcun modo, alterarne il comportamento e, soprattutto, aggiungere nuove funzionalità.
+Se l'attività è stata fatta bene, al termine avremo un codice più pulito, comprensibile, di cui non è stato alterato il comportamento e che passa perfettamente tutti i test già presenti.
+Ma andiamo per gradi.
+
+## Benefici
+
+Uno dei principali benefici che otterremo con il refactoring è la rimozione di porzioni di codice non performanti, sostituiti da codice più trasparente e chiaro. E con questo, un'evidente riduzione del debito tecnico accumulato.
+Un codice scritto bene, non solo aiuta a chi lo ha scritto di poterlo rivedere a distanza di qualche tempo senza troppi mal di testa ma, con il crescere del software e l'introduzione di nuove funzionalità, rende più semplice la vita ai futuri sviluppatori che dovranno interfacciarsi con esso.
+Da un codice scritto bene, si possono ottenere migliori performance. Magari non direttamente ma, una buona attività di refactoring ci permetterà di avere molto più chiaro come il software interagisce, dandoci la possibilità di individuare eventuali colli di bottiglia, anche a livello architetturale.
+
+## Quando
+
+La scrittura del codice non è del tutto lineare, soprattutto se scritto a più mani. Ogni sviluppatore ha un suo stile, delle sue convinzioni, un determinato livello di preparazione. Questo porta spesso ad avere del codice, scritto a colpi di commit, funzionante ma difficile da comprendere, con scorciatoie evitabili, metodi prolissi o, peggio ancora, costanti che non dovrebbero esserlo.
+Sono diverse le occasioni in cui è possibile fare refactoring e, in alcune di queste, non solo è consigliato ma obbligatorio.
+Poniamo il caso che abbiamo una parte del nostro software con del codice "legacy", ovvero scritto più di tre mesi fà, che funziona egregiamente, fa il suo ma, ci rendiamo conto che le performance non sono quelle che ci attendiamo.
```

Comment:
Non sono molto d'accordo con l'idea che "legacy" voglia dire vecchio, e sicuramente 3 mesi è un tempo molto opinabile. Che ne pensi di rivederlo in ottica di "codice che non rispetti determinati standard che abbiamo imposto" o qualcosa di simile?

Reply:
Sì, son daccordo. Voleva solo essere un modo per sottolineare che il codice è vivo e che il ciclo di vita a volte è molto breve. Ma mi trovi daccordo.

Reply:
Absolutely, mi torna, però ricordiamoci sempre il target: Qualsiasi dev da qualsiasi astrazione e livello di esperienza e un numero, o una metrica di qualsiasi tipo, potrebbe essere presa come dato di fatto :)

Reply:
Rivisto in modo da sottrarre importanza al mero concetto di performance per darne al rispetto dei principi SOLID e ai pattern non rispettati